client/rest-high-level/src/main/java/org/elasticsearch/client/security/ClearSecurityCacheRequest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.Validatable;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * The request used to clear the specified cache.
+ */
+public final class ClearSecurityCacheRequest implements Validatable {
+
+    private final String cacheName;
```

Comment:
I think string is fine since there is a `CacheInvalidator` that already controls the available/valid entries. I don't see what extra value enum could provide here. Please also see my other [comment](https://github.com/elastic/elasticsearch/pull/59376#discussion_r499199071) about why it is a request param.

Reply:
The construction of `ClearSecurityCacheRequest` object happens on the client side. We might have validation for its parameters on the server side, but the caller has to hard code the "api_key" value in the client code. Something like:
```
public final class ClearSecurityCacheRequest implements Validatable {
    
    enum CacheName {
        API_KEY
    }

    private final CacheName cacheName;
```
would help a bit.

A separate request object type would still be preferable IMO.

Reply:
Client side is a good point. Until we can have auto-generation of client from API spec, we'll have to maintain this manually. I added a dedicated `ClearApiKeyCacheRequest` for the client code.